ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/TimeoutParticipantTestContext.scala

```diff
@@ -0,0 +1,497 @@
+  override def submit(request: SubmitRequest): Future[Unit] =
+    withTimeout(s"Submit for request $request", delegate.submit(request))
+  override def submitAndWait(request: SubmitAndWaitRequest): Future[Unit] = withTimeout(
+    s"Submit and wait for request $request",
+    delegate.submitAndWait(request),
+  )
+  override def submitAndWaitForTransactionId(
+      request: SubmitAndWaitRequest
+  ): Future[SubmitAndWaitForTransactionIdResponse] = withTimeout(
+    s"Submit and wait for transaction id for request $request",
+    delegate.submitAndWaitForTransactionId(request),
+  )
+  override def submitAndWaitForTransaction(
+      request: SubmitAndWaitRequest
+  ): Future[SubmitAndWaitForTransactionResponse] = withTimeout(
+    s"Submit and wait for transaction for request $request",
+    delegate.submitAndWaitForTransaction(request),
+  )
+  override def submitAndWaitForTransactionTree(
+      request: SubmitAndWaitRequest
+  ): Future[SubmitAndWaitForTransactionTreeResponse] = withTimeout(
+    s"Submit and wait for transaction tree for request $request",
+    delegate.submitAndWaitForTransactionTree(request),
+  )
+  override def completionStreamRequest(from: LedgerOffset)(
+      parties: Primitive.Party*
+  ): CompletionStreamRequest = delegate.completionStreamRequest(from)(parties: _*)
+  override def completionEnd(request: CompletionEndRequest): Future[CompletionEndResponse] =
+    withTimeout(
+      s"Completion end for request $request",
+      delegate.completionEnd(request),
+    )
+  override def completionStream(
+      request: CompletionStreamRequest,
+      streamObserver: StreamObserver[CompletionStreamResponse],
+  ): Unit = delegate.completionStream(request, streamObserver)
+  override def firstCompletions(request: CompletionStreamRequest): Future[Vector[Completion]] =
+    withTimeout(
+      s"First completions for request $request",
+      delegate.firstCompletions(request),
+    )
+  override def firstCompletions(parties: Primitive.Party*): Future[Vector[Completion]] =
+    withTimeout(
+      s"First completions for parties $parties",
+      delegate.firstCompletions(parties: _*),
+    )
+  override def findCompletionAtOffset(offset: HexString, p: Completion => Boolean)(
+      parties: Primitive.Party*
+  ): Future[Option[ParticipantTestContext.CompletionResponse]] = withTimeout(
+    s"Find completion at offset $offset for parties $parties",
+    delegate.findCompletionAtOffset(offset, p)(parties: _*),
+  )
+  override def findCompletion(request: CompletionStreamRequest)(
+      p: Completion => Boolean
+  ): Future[Option[ParticipantTestContext.CompletionResponse]] = withTimeout(
+    s"Find completion for request $request",
+    delegate.findCompletion(request)(p),
+  )
+  override def findCompletion(parties: Primitive.Party*)(
+      p: Completion => Boolean
+  ): Future[Option[ParticipantTestContext.CompletionResponse]] =
+    delegate.findCompletion(parties: _*)(p)
```

Comment:
Do we need a timeout here?

Reply:
Yes, good catch, my brain died adding all this boilerplate

Reply:
Yeah, I feel you, had to review in parts too 😄